TestingExamples/TestingExamples/DataValidationTests/ResponseBodyTesting.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Net;
+using FluentAssertions;
+using NUnit.Framework;
+using RestSharp;
+using Services.Models;
+using Services.RequestModels;
+
+namespace ApiTestingExamples.DataValidationTests;
+
+public class ResponseBodyTesting : BaseTest
+{
+    [Test]
+    [Property("Test", "DataValidationTesting")]
+    public async Task ResponseBodyTesting_DataMatchesExpectedValues()
+    {
+        await ValidateResponseData("TestName", 42, DateTime.Now);
+    }
+
+    private async Task ValidateResponseData(string dataName, int dataIndicator, DateTime endDateTime)
+    {
+        // ARRANGE
+        var request = new RestRequest($"/api/testing/return/the/data/endpoint")
+            .AddJsonBody(new DataRequest
+            {
+                StartDate = endDateTime.AddMinutes(-15),
+                EndDate = endDateTime,
+                DataName = dataName,
+                DataIndicator = dataIndicator
+            });
+
+        // ACT
+        var response = await AuthorizedRestClient.ExecutePostAsync<GetDataRequestResponse>(request);
+
+        // ASSERT
+        response.Should().NotBeNull();
+        response.StatusCode.Should().Be(HttpStatusCode.OK);
+
+        response.Data?.StructuredData.Any(x => x.Id == dataIndicator).Should().BeTrue();
+        response.Data?.StructuredData.Any(x => x.Name == dataName).Should().BeTrue();
```

Comment:
Should you be validating this against the dataName you pass in instead of just validating that it exists?

Reply:
Ok I updated the way this was handled to do a select and then should be equal or equivalent to depending on data type